recipe/run_test.sh

```diff
@@ -7,14 +7,24 @@ pushd "tests"
 
 if [[ $PKG_NAME == "openmpi" ]]; then
 
-  if [[ -n "$(conda list | grep ucx)" ]]; then
-    echo "Improper UCX dependency!"
-    exit 1
```

Comment:
We need to keep the old tests. The purpose is different from the new ones that you added:
- `conda list` tests: ensure the optional dependencies are not installed by default
- `ompi_info` tests: ensure the support for optional dependencies was built

Reply:
I find it much more relevant to verify that OpenMPI has effectively been built with the desired features (here UCX and CUDA).
This approach is more comprehensive because it goes beyond verifying the mere presence of dependencies and checks the actual runtime configuration and capabilities of the package, providing stronger assurance that the package will function correctly in the user's environment.